velox/functions/lib/Re2Functions.cpp

```diff
@@ -594,27 +648,52 @@ class LikeGeneric final : public VectorFunction {
       const auto reducedLength = patternMetadata.length;
       const auto& fixedPattern = patternMetadata.fixedPattern;
 
-      switch (patternMetadata.patternKind) {
-        case PatternKind::kExactlyN:
-          return OptimizedLike<PatternKind::kExactlyN>::match(
-              input, pattern, reducedLength);
-        case PatternKind::kAtLeastN:
-          return OptimizedLike<PatternKind::kAtLeastN>::match(
-              input, pattern, reducedLength);
-        case PatternKind::kFixed:
-          return OptimizedLike<PatternKind::kFixed>::match(
-              input, fixedPattern, reducedLength);
-        case PatternKind::kPrefix:
-          return OptimizedLike<PatternKind::kPrefix>::match(
-              input, fixedPattern, reducedLength);
-        case PatternKind::kSuffix:
-          return OptimizedLike<PatternKind::kSuffix>::match(
-              input, fixedPattern, reducedLength);
-        case PatternKind::kSubstring:
-          return OptimizedLike<PatternKind::kSubstring>::match(
-              input, fixedPattern, reducedLength);
-        default:
-          return applyWithRegex(input, pattern, escapeChar);
+      if (isAscii) {
```

Comment:
Can we move this check under case PatternKind::kExactlyN and PatternKind::kAtLeastN? Currently, there is a lot of copy-paste that makes the code hard to read.

Reply:
Got it. This is fine. Appreciate the explanation.